### PR TITLE
We switch on `network` to determine signatories for our `MultiSigWallet`

### PR DESCRIPTION
**Pull request by saturnial**
- We read in values from an external file to determine who are the signatories for our `MultiSigWallet`

### CODE REVIEW

File: migrations/2_deploy_dharma_contracts.js [hunk 1]
```diff
@@ -1,22 +1,44 @@
...
     const MultiSigWallet = artifacts.require("MultiSigWallet");
 
+    // We switch on the network to ensure we're configuring our MultiSigWallet
+    // accordingly.
+    let owners;
+    switch (network) {
+        case "live":
+            owners = OWNERS.owners;
+            break;
+        case "kovan":
```
Comment (nadavhollander) [inline migrations/2_deploy_dharma_contracts.js]: Are we confident these are the names truffle uses to represent kovan / testrpc?

File: migrations/2_deploy_dharma_contracts.js [hunk 2]
```diff
@@ -1,22 +1,44 @@
+const OWNERS = require("./OWNERS");
```
Comment (nadavhollander) [inline migrations/2_deploy_dharma_contracts.js]: Let's clarify the naming here -- i.e. `OWNERS` => something like `MULTISIG_SIGNATORIES`

File: migrations/2_deploy_dharma_contracts.js [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
+            break;
+        case "kovan":
+        case "development":
+            owners = accounts;
+            break;
+        default:
+            throw "invalid network";
+    }
+
+    const required = Math.floor(owners.length / 2);
```
Comment (nadavhollander) [inline migrations/2_deploy_dharma_contracts.js]: Ideally would like to parametrize this somehow as well -- i.e. have the `threshold` specified in the `OWNERS.js`/`MULTISIG_SIGNATORIES.js` file.

File: migrations/2_deploy_dharma_contracts.js (hunk 2 above)
Comment (saturnial) [reply migrations/2_deploy_dharma_contracts.js]: Do you want to change the filename, the variable name or both?

File: migrations/2_deploy_dharma_contracts.js (hunk 1 above)
Comment (saturnial) [reply migrations/2_deploy_dharma_contracts.js]: Do you know otherwise? Link?

File: migrations/2_deploy_dharma_contracts.js (hunk 3 above)
Comment (saturnial) [reply migrations/2_deploy_dharma_contracts.js]: Can do!

File: migrations/2_deploy_dharma_contracts.js (hunk 2 above)
Comment (nadavhollander) [reply migrations/2_deploy_dharma_contracts.js]: Both please :)

File: migrations/2_deploy_dharma_contracts.js (hunk 1 above)
Comment (saturnial) [reply migrations/2_deploy_dharma_contracts.js]: We can also just special case the `live` network and assign the `accounts` for everything else

Comment (nadavhollander) [reply migrations/2_deploy_dharma_contracts.js]: I don't -- just wasn't totally sure how the `network` string gets assigned.

Also, this is a bit of nit, but I'd prefer if string literals were subbed in for constant -- i.e. `const LIVE_NETWORK_ID = "live";` etc.

File: migrations/2_deploy_dharma_contracts.js (hunk 3 above)
Comment (nadavhollander) [reply migrations/2_deploy_dharma_contracts.js]: 🎸

File: migrations/2_deploy_dharma_contracts.js (hunk 1 above)
Comment (nadavhollander) [reply migrations/2_deploy_dharma_contracts.js]: Yeah, i think special casing just `"live"` make sense.

Comment (saturnial) [reply migrations/2_deploy_dharma_contracts.js]: Not a nit at all, let's definitely make a constant.

**Pull request merged**